django_auth_ldap/backend.py

```diff
@@ -1002,6 +1001,14 @@ def __init__(self, prefix='AUTH_LDAP_', defaults={}):
             value = getattr(django.conf.settings, prefix + name, default)
             setattr(self, name, value)
 
+        # Compatibility with old caching settings: if CACHE_TIMEOUT isn't changed and CACHE_GROUPS is True
+        if getattr(django.conf.settings, prefix + 'CACHE_GROUPS', defaults.get('CACHE_GROUPS')) \
+                and self.CACHE_TIMEOUT == self.defaults['CACHE_TIMEOUT']:
+            # Previously GROUP_CACHE_TIMEOUT defaulted to None, which means cache forever. That's not a good default,
+            # we now fall back to 1 hour. See https://github.com/django-auth-ldap/django-auth-ldap/issues/19
+            self.CACHE_TIMEOUT = getattr(django.conf.settings, prefix + 'GROUP_CACHE_TIMEOUT',
+                                         defaults.get('GROUP_CACHE_TIMEOUT', 3600))
```

Comment:
Please emit a warning when `CACHE_GROUPS` or `GROUP_CACHE_TIMEOUT` is exists. It should be considered deprecated and users should upgrade to the new `CACHE_TIMEOUT` setting. To emit a warning, you can use the [`warnings`](https://docs.python.org/3/library/warnings.html) module. Something like:

```py
warnings.warn(...)
```

---

We'll eventually drop `CACHE_GROUPS` and `GROUP_CACHE_TIMEOUT`, so I think I'd prefer to keep the old default for backwards compatibility. The new setting does not have this issue. Can you keep the default of `None` for `GROUP_CACHE_TIMEOUT`, please?